src/util/server_table.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * server_table.c
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <pelz_io.h>
+#include <server_table.h>
+#include <util.h>
+#include <pelz_request_handler.h>
+#include <charbuf.h>
+#include <pelz_log.h>
+
+#include "sgx_trts.h"
+#include "pelz_enclave_t.h"
+#include "kmyth_enclave.h"
+
+ServerTable server_table;
+
+//Initialize server table
+int server_table_init(void)
+{
+  if ((server_table.entries = (CertEntry *) malloc(sizeof(CertEntry))) == NULL)
+  {
+    pelz_log(LOG_ERR, "Server List Space Allocation Error");
+    return (1);
+  }
+
+  server_table.num_entries = 0;
+  server_table.mem_size = 0;
+  return (0);
+}
+
+//Destroy server table
+int server_table_destroy(void)
+{
+  pelz_log(LOG_DEBUG, "Server Table Destroy Function Starting");
+  if (server_table.num_entries >= 0)
```

Comment:
```server_table.num_entries``` is always non-negative, so you always follow this path and never execute the ```else``` statement. This should check that ```server_table``` has been initialized and free entries if it's number of entries is positive.